### PR TITLE
[alea] Add test for operator== and operator!=

### PR DESCRIPTION
**Pull request by mwallerb**
Fixes: #482

### CODE REVIEW

**Comment by egull** [comment]
@galexv this is in response to one of your comments, please have a look at it.

File: alea/test/compare.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    void test_one()
+    {
+        Acc acc1(2), acc2(2);
+
+        fill(acc1, 0, 1);
+        EXPECT_NE(acc1.result(), acc2.result());
+
+        acc2 = acc1;
+        EXPECT_EQ(acc1.result(), acc2.result());
```
Comment (galexv) [inline alea/test/compare.cpp]: We do rely here on the fact that assignment operator makes objects equal.
Or, in other words, "equal objects are, _by definition_, those which are made by an assignment operator".

Markus noted that IEEE FP standard does not even guarantee that two results made from identical accumulators will be strictly numerically equal --- but if they are indeed not equal on some specific platform and this test fails, we would like to know about it.

Comment (mwallerb) [reply alea/test/compare.cpp]: Good point, thanks for making it explicit.
The test is mainly supposed to take of two corner cases:
 - 0 data points, where the mean does not exist
 - 1 data point, where the variance does not exist
